cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -454,6 +462,191 @@ struct PowerChecked {
   }
 };
 
+struct Sin {
+  template <typename T, typename Arg0>
+  static enable_if_integer<Arg0, T> Call(KernelContext*, Arg0 val, Status*) {
```

Comment:
As mentioned elsewhere, I don't think it's useful to expose integer-accepting versions of these kernels.

Reply:
I've gone ahead and removed them.